### ISSUE TITLE
[REQ] Restructure mods folder

### ISSUE THREAD

**Issue by BizzarBlitz** [pizzaboxer/bloxstrap#2788]
### Acknowledgement of preliminary instructions

- [X] I have read and acknowledged the preliminary instructions.

### What idea do you have?

# TLDR
Go from this:
```
Mods
├ ClientSettings
├ content
└ ExtraContent
```
to this:
```
Mods
├ Bloxstrap
│ ├ ClientSettings
│ ├ content
│ └ ExtraContent
├ Mod1
│ ├ ClientSettings
│ ├ content
│ └ ExtraContent
├ Mod2
│ ├ ClientSettings
│ ├ content
│ └ ExtraContent
```
----
Currently, mods are all grouped into one `Modifications` folder. This makes uninstalling/updating a specific mod a challenge when it shares directories with another. I suggest separating each mod into its own folder (as seen above), which would not only separate mods visually but also make them easy to uninstall (just delete the specific mod folder).

### Bloxstrap
Bloxstrap would get its own `Bloxstrap` mod folder, which, on top of making program modifications distinguishable from user modifications, also causes less headaches/confusion for users wondering why their mods are being overwritten.

### Conflicting Mods
Conflicting mods could be handled in two ways:
1. Bloxstrap warns the user that they have conflicting mod folders and arbitrarily picks one over the other
2. A new priority system is added (out of scope for this suggestion though)
If any user mods conflict with the `Bloxstrap` mod folder, then the users should always take priority

### Possibilities
Although out of the scope of this suggestion, separating mod folders would open the door for other features, such as:
* Disabling specific mods
* A "run bloxstrap without user mods" option
* Mods with their own fflags (via different `ClientSettings` files)
* [An "install mod" button](https://discord.com/channels/1099468797410283540/1145288956297093130/1145288956297093130)

([Original Discord suggestion](https://discord.com/channels/1099468797410283540/1200191299161182329))

**Comment by Storm-930** [comment]
what? an actual suggestion? in this day and age? absolutely frivulous! imposible!

**Comment by BlueOutside** [comment]
TLDR of the TLDR
Use the AppleBlox mods structure 😊

**Comment by pizzaboxer** [comment]
So I was already thinking of a potential way to handle mods in an encapsulated/containerized way, but this definitely seems to be a much better way of doing things, and compliments what I have in mind.

**Comment by ShadowCodeX-debug** [comment]
This can go hand-to-hand with #1098

**Comment by MistressDoom** [comment]
thank you
